Mathlib/Algebra/Order/Group/Defs.lean

```diff
@@ -42,6 +43,25 @@ class OrderedCommGroup (α : Type u) extends CommGroup α, PartialOrder α where
   protected mul_le_mul_left : ∀ a b : α, a ≤ b → ∀ c : α, c * a ≤ c * b
 #align ordered_comm_group OrderedCommGroup
 
+/-- An ordered (additive) commutative group *with one* is an additive commutative group with one
+with a partial order such that `a ≤ b → c + a ≤ c + b` (addition is monotone) and `0 ≤ 1`. -/
+class OrderedAddCommGroupWithOne (α : Type u) extends AddCommGroupWithOne α, PartialOrder α where
+  /-- Addition is monotone in an `OrderedAddCommGroupWithOne`. -/
+  protected add_le_add_left : ∀ a b : α, a ≤ b → ∀ c : α, c + a ≤ c + b
+  /-- One is nonnegative in an `OrderedAddCommGroupWithOne`. -/
```

Comment:
Referring to the specific structure `A` in the docstrings of fields of `A` is nonsense to me, because when later `B` extends `A`, the docstrings of some fields of `B` (the ones inherited from `A`) will refer to `A`.
Regardless, those docstrings add no value since they rephrase properties that are easy to read in the first place. What about you just don't write them? If the linter complains, silence it: It's because it erroneously thinks `add_le_add_left` is a `def`.

Reply:
This is how we have written other existing docstrings (including ones in this file), and I don't really see a more sensible option. Note that docstrings are required on *all* structure fields, regardless of whether they are `Prop`s, and that's not (I don't think, but I could be wrong if you point me to the source) because it thinks `add_le_add_left` is a `def`. Unless the linter is *actually wrong* about something, I'm hesitant to just silence it.

So for those reasons, I'm leaving this as is.

Reply:
The linter is acting based on incorrect behavior in lean core (of using `def` for proofs). It is totally fine to `nolint` these fields if you thing a docstring is silly.

Reply:
Aha, I was wrong then. I think I will leave them anyway only to match the other declarations in this file (and other files). If we decide we don't want them later, we can probably write something to mass delete them.